okhttp/src/main/java/okhttp3/internal/http2/Hpack.java

```diff
@@ -152,7 +154,6 @@ private void adjustDynamicTableByteCount() {
     }
 
     private void clearDynamicTable() {
-      headerList.clear();
```

Comment:
I triggered an edge case here in the HPACK decoder. If the server HPACK encoder sent a header intended to be added to the dynamic table, and it's length was larger than the entire dynamic table, we'd clear the header list. My understanding is that we should never clear the header list, even if a header wipes out all entries in the dynamic table.

To see it in action, re-add this line and run HpackTest#largeHeaderValue test case.